apps/apps.c

```diff
@@ -686,18 +693,45 @@ X509_CRL *load_crl(const char *infile, int format)
     } else if (format == FORMAT_PEM) {
         x = PEM_read_bio_X509_CRL(in, NULL, NULL, NULL);
     } else {
-        BIO_printf(bio_err, "bad input format specified for input crl\n");
+        print_format_error(format, OPT_FMT_PEMDER);
         goto end;
     }
-    if (x == NULL) {
-        BIO_printf(bio_err, "unable to load CRL\n");
+
+ end:
+    if (x == NULL && desc != NULL) {
+        BIO_printf(bio_err, "unable to load %s\n", desc);
         ERR_print_errors(bio_err);
         goto end;
```

Comment:
What I meant with the comment above: doesn't that look like an endless loop?

Reply:
Ah, this was a bad copy&paste mistake of mine, thanks. Removed.